### PR TITLE
cmake: fix non portable code

### PR DESCRIPTION
**Pull request by selsta**
Co-authored-by: @perfect-daemon

Required for e.g. Xcode or ninja msys2

### CODE REVIEW

File: src/blocks/CMakeLists.txt [hunk 1]
```diff
@@ -28,20 +28,32 @@
...
+string(REGEX REPLACE "[0-9a-fA-F][0-9a-fA-F]" "0x\\0," DATA "${DATA}")
+file(WRITE "${OUTPUT_C_SOURCE}" "
+#include <stddef.h>
+const unsigned char ${BLOB_NAME}[]={
+  ${DATA}
+};
+const size_t ${BLOB_NAME}_len = sizeof(${BLOB_NAME});
+"
+)
+]=])
```
Comment (xiphon) [inline src/blocks/CMakeLists.txt:43]: Why do you need to dynamically generate this `.cmake` file?

Comment (selsta) [reply src/blocks/CMakeLists.txt:43]: No need for it but one less file in source tree. It does not matter really.

Comment (mj-xmr) [reply src/blocks/CMakeLists.txt:43]: I would personally vote for a file in source tree. We are mixing syntax of two languages here.

Comment (selsta) [reply src/blocks/CMakeLists.txt:43]: We will mix two languages anyway.

**Review by mj-xmr — APPROVED** [review]
Actually when I look at it again, it appears much cleaner than the previous version.

**Pull request merged**